homeassistant/components/tasmota/update.py

```diff
@@ -0,0 +1,79 @@
+"""Update entity for Tasmota."""
+
+import re
+
+from homeassistant.components.update import (
+    UpdateDeviceClass,
+    UpdateEntity,
+    UpdateEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.device_registry import DeviceEntry
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .coordinator import TasmotaLatestReleaseUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up Tasmota update entities."""
+    coordinator = TasmotaLatestReleaseUpdateCoordinator(hass, config_entry)
+    await coordinator.async_config_entry_first_refresh()
+
+    device_registry = dr.async_get(hass)
+    devices = device_registry.devices.get_devices_for_config_entry_id(
+        config_entry.entry_id
+    )
+    async_add_entities(TasmotaUpdateEntity(coordinator, device) for device in devices)
+
+
+class TasmotaUpdateEntity(UpdateEntity):
```

Comment:
I'm not sure if we should do things here a bit differently (and if, how) in order to:
- do the right thing when the firmware of the device gets updated behind the scenes (unsure if we should add an async_update method that fetches a new device registry entry every time)
- set up the update entity on the fly for newly added Tasmota devices